src/templates/promotions/discounts/_edit.html

```diff
@@ -185,19 +185,38 @@
                 </div>
             {% endblock %}
         {% endembed %}
+        
+        {% set usersInput %}
+            <div class="flex">
+                <div>
+                    {{ forms.select({
+                        id: 'user-condition',
+                        name: 'userCondition',
+                        options: discount.getUserConditions(),
+                        value: discount.userCondition
+                    }) }}
+                </div>
+        
+                <div class="flex-grow{% if discount.userCondition is null or discount.userCondition == 'usersAnyOrNone' %} hidden{% endif %}">
+                    {{ forms.multiselectField({
+                        id: 'groups',
+                        name: 'groups',
+                        options: groups,
+                        values: discount.getUserGroupIds(),
+                        errors: discount.getErrors('groups'),
+                        class: 'selectize fullwidth',
+                    }) }}
+                </div>
+            </div>
+        {% endset %}
 
-        {% if groups|length %}
-            {{ forms.multiselectField({
-                label: 'User Groups'|t('commerce'),
-                instructions: "Groups for which this discount will be applicable to. Leave blank for all groups"|t('commerce'),
-                id: 'groups',
-                name: 'groups',
-                options: groups,
-                values: discount.getUserGroupIds(),
-                errors: discount.getErrors('groups'),
-                class: 'selectize fullwidth',
-            }) }}
-        {% endif %}
+        {{ forms.field({
+            id: 'user-condition',
+            label: 'User Condition'|t('commerce'),
+            instructions: 'Choose a condition for user restriction.'|t('commerce'),
```

Comment:
If it makes sense, I’d propose changing this to:

> Choose which user groups should match this discount.

That better frames “Any or none” since it will apply by default and not have any limiting impact, and makes it easier to consider the additional options.